sub-packages/bionemo-geometric/tests/bionemo/geometric/test_molecule_featurizers.py

```diff
@@ -0,0 +1,483 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: LicenseRef-Apache2
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import re
+
+import pytest
+import torch
+from rdkit import Chem
+from rdkit.Chem import Descriptors
+
+from bionemo.geometric.molecule_featurizers import RDkit2DDescriptorFeaturizer
+
+
+@pytest.fixture(scope="module")
+def sample_mol():
+    return Chem.MolFromSmiles("NC(=O)c1cn(-c2ccc(S(N)(=O)=O)cc2)nc1-c1ccc(Cl)cc1")  # CHEMBL3126825
+
+
+@pytest.fixture(scope="module")
+def sample_mol2():
+    return Chem.MolFromSmiles("C[C@H]1CN(c2ncnc3[nH]cc(-c4cccc(F)c4)c23)CCO1")  # CHEMBL3927167
+
+
+def test_rdkit2d_descriptor_featurizer(sample_mol, sample_mol2):
+    rdf = RDkit2DDescriptorFeaturizer()
+    mol_feats = rdf(sample_mol)
+
+    # separate out int and float descriptors
+    int_desc_idx = [
+        idx for idx, (name, _) in enumerate(Descriptors.descList) if re.search(r"(num|fr_|count)", name, re.IGNORECASE)
+    ]
+    float_desc_idx = list(set(range(len(Descriptors.descList))) - set(int_desc_idx))
+
+    # 2D RDkit descriptors listed in rdkit.Chem.Descriptors.descList
+    mol_feats_ref = torch.Tensor(
```

Comment:
Is there a way to generate the reference on the fly so that you don't have to hardcode it here?